test/clients/services_manager_test.rb

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative '../test_helper'
+
+require "y2journal"
+
+require "yast"
+require "services-manager/clients/services_manager"
+
+describe Y2ServicesManager::Clients::ServicesManager do
+  include Yast::UIShortcuts
+
+  # Checks whether a widgets tree contains the given id
+  #
+  # @param tree [Yast::Term]
+  # @param id [Symbol]
+  #
+  # @return [Boolean]
+  def contain_widget?(tree, id)
+    !find_widget(tree, value: :id, param: id).nil?
+  end
+
+  # Finds a widget in the widgets tree
+  #
+  # @param tree [Yast::Term]
+  # @param value [Symbol]
+  # @param param [Object]
+  #
+  # @return [Yast::Term, nil]
+  def find_widget(tree, value: nil, param: nil)
+    return nil unless tree.is_a?(Yast::Term)
+
+    tree.nested_find do |widget|
+      widget.is_a?(Yast::Term) &&
+        widget.value == value &&
+        widget.params.any?(param)
+      end
+  end
+
+  # Checks whether the widgets tree contains a button with a specific label
+  #
+  # @param tree [Yast::Term]
+  # @param label [String]
+  #
+  # @return [Boolean]
+  def contain_button?(tree, label)
+    !find_widget(tree, value: :PushButton, param: label).nil?
+  end
+
+  # Checks whether the widgets tree contains a menu button with a specific label and
+  # options (optional)
+  #
+  # @param tree [Yast::Term]
+  # @param label [String]
+  # @param options [Array<Symbol>]
+  #
+  # @return [Boolean]
+  def contain_menu_button?(tree, label, options: [])
+    widget = find_widget(tree, value: :MenuButton, param: label)
+
+    return false if widget.nil?
+
+    options.all? do |option|
+      widget.params.last.any? { |opts| contain_widget?(opts, option) }
+    end
+  end
+
+  # Helper for buttons expectations
+  #
+  # @param block [Proc]
+  def expect_buttons_to(&block)
```

Comment:
well, names does not match well what it did. It looks it expect refresh of buttons with some content?